redash/handlers/queries.py

```diff
@@ -171,6 +171,17 @@ def get(self):
 
         return response
 
+def require_access_to_dropdown_queries(user, query_def):
+    parameters = query_def.get('options', {}).get('parameters', [])
+    dropdown_query_ids = [str(p['queryId']) for p in parameters if p['type'] == 'query']
+
+    if dropdown_query_ids:
+        groups = models.db.session.execute('select group_id, view_only from queries join data_source_groups on queries.data_source_id = data_source_groups.data_source_id where queries.id in ({})'.format(','.join(dropdown_query_ids))).fetchall()
```

Comment:
The concept is definitely better than previously: we run a single query and don't have to do N+1 queries twice. 👌 

But the implementation has a few issues -- 

1. All direct queries should be in `redash.models`. Outside of `redash.models` we should  use an API exposed by the models.
2. By building a query on your own, you open this to SQL injections. I'm pretty sure `session.execute` can take a parameterized query and if not there is a similar API that can do this.
3. Why not use SQLAlchemy's models to build this query? I know it's more effort at first due to the learning curve, but it's more consistent with the rest of the code and has other benefits.

Reply:
1. Would you feel better with this hanging off a `Query` class method?
2. You're right, should have used params.
3. I saw this coming, but decided to push a quicker fix to hear your thoughts, specifically because
> If the fix isn't that complex, worth handling

Reply:
1. Yes. Something like `Query.get_groups_for_query_ids`?
3. 👍

Reply:
d8339f3